src/library_fs.js

```diff
@@ -726,6 +726,9 @@ FS.staticInit();` +
         lookup = FS.lookupPath(new_path, { parent: true });
         new_dir = lookup.node;
       } catch (e) {
+        // do not override failing error
```

Comment:
What if we just remove this try/catch completely?

I guess we should also a test (or modify an existing test)?

Reply:
Do not know what that will do if we remove it for other error messages.  Just let me know if you would prefer to remove the try catch.

What tests are there for that and where so I can change it.

Reply:
There are tests as part of the file system tests that this fixes.

Reply:
It doesn't look like we have a test for returning EBUSY here - nothing comes up in a search, and removing the try-catch as @sbc100 suggests passes the test suite.

Reading the docs on `rename()`, I don't think we should be returning EBUSY. So I think that's the proper fix, unless we have other information?

Reply:
Looks like this failure case should probably be added to `tests/stdio/test_rename.c`

Reply:
So remove the try catch and just let it fail and pass the error percolate up is the way to fix this right.

Reply:
To me that look like a better solution yes... I can't see why use the EBUSY thing is below...  but I don't know this code.

Best thing is to upload that patch and see if any tests fail.

Reply:
Will do.  Thanks

Reply:
Addressed in recent commit